### PR TITLE
GitHub Actions fixes: respect lock file, unify CI, only publish once

### PR DESCRIPTION
**Pull request by cakoose**
1. Use `npm ci` instead of `npm install` to respect the package versions in the lockfile.
2. Merge "feature.yml" into "main.yml". They were doing the same thing.
3. For "publish.yml", only use one Node.js version, since we don't want to publish multiple times. Also no need to run tests.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm install
```
Comment (carterzenk) [inline .github/workflows/main.yml:20]: Should we make the same change on `publish.yml` and `feature.yml`?

Comment (cakoose) [reply .github/workflows/main.yml:20]: Oh, I didn't even realize there was another `.yml` file. That's probably why the Prettier workflow only failed after the PR landed.

The workflows have the same contents, so lemme just merge them.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -5,18 +5,14 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
```
Comment (cakoose) [inline .github/workflows/publish.yml:8]: We don't want to run the publish step multiple times, so just one one Node.js version.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -2,8 +2,6 @@ name: Lint, build & test
 
 on:
   push:
-    branches:
```
Comment (cakoose) [inline .github/workflows/main.yml:5]: Merged the `on:` sections of "main.yml" and "feature.yml" since they were doing the same thing anyway.

**Review by carterzenk — APPROVED** [review]

**Pull request merged**